db/tasks.go

```diff
@@ -176,8 +176,7 @@ func (db *DB) GetTask(id uuid.UUID) (*Task, error) {
 	return r[0], nil
 }
 
-func (db *DB) CreateBackupTask(owner string, job *Job) (*Task, error) {
-	id := uuid.NewRandom()
+func (db *DB) CreateBackupTask(owner string, job *Job, id uuid.UUID) (*Task, error) {
```

Comment:
If you are not "forcibly impose a UUID" then it is not possible to return back to the caller a ID.
This is running on the jobapi part, which sends  the task over a channel which supervisor. The task is is created on the other side of the channel. Do you think another solution to have available the taskid?

Agree that every caller to createBackuTask has to do uuid.NewUuid() which is not optimal.